src/Middlewares/CacheResponse.php

```diff
@@ -25,15 +26,31 @@ public function handle(Request $request, Closure $next, $lifetimeInSeconds = nul
             if ($this->responseCache->hasBeenCached($request)) {
                 event(new ResponseCacheHit($request));
 
-                return $this->responseCache->getCachedResponseFor($request);
+                $response = $this->responseCache->getCachedResponseFor($request);
+
+                collect(config('responsecache.replacers', []))->map(function ($replacerClass) {
+                    return resolve($replacerClass);
+                })->each(function (Replacer $replacer) use ($response) {
+                    $replacer->replaceCachedResponse($response);
+                });
+
+                return $response;
             }
         }
 
         $response = $next($request);
 
         if ($this->responseCache->enabled($request)) {
             if ($this->responseCache->shouldCache($request, $response)) {
-                $this->responseCache->cacheResponse($request, $response, $lifetimeInSeconds);
+                $cachedResponse = clone $response;
```

Comment:
Maybe put this in its own dedicated function. Could be called `makeReplacementsAndStoreResponse`